cuML/src/metrics/trustworthiness.cu

```diff
@@ -0,0 +1,162 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "trustworthiness.h"
+#include <cuda_utils.h>
+#include "distance/distance.h"
+#include <selection/columnWiseSort.h>
+#include <common/cumlHandle.hpp>
+#include "../knn/knn.h"
+
+using namespace MLCommon;
+using namespace MLCommon::Selection;
+using namespace ML;
+
+namespace ML {
+
+    /**
+    * @brief Compute a kNN and returns the indexes of the nearest neighbors
+    * @input param input: Input matrix holding the dataset
+    * @input param n: Number of samples
+    * @input param d: Number of features
+    * @return Matrix holding the indexes of the nearest neighbors
+    */
+    template<typename math_t>
+    long* get_knn_indexes(const cumlHandle& h, math_t* input, int n,
+                                int d, int n_neighbors)
+    {
+        cudaStream_t stream = h.getStream();
+        auto alloc = h.getHostAllocator();
+        
+        long* d_pred_I;
+        math_t* d_pred_D;
+        allocate<long>(d_pred_I, n * n_neighbors);
+        allocate(d_pred_D, n * n_neighbors);
+
+        kNNParams params = {input, n};
+        kNN knn(d);
+        knn.fit(&params, 1);
+        knn.search(input, n, d_pred_I, d_pred_D, n_neighbors);
+
+        long* h_pred_I = (long*)alloc->allocate(n * n_neighbors * sizeof(long), stream);
+        updateHost(h_pred_I, d_pred_I, n * n_neighbors, stream);
+
+        CUDA_CHECK(cudaFree(d_pred_I));
+        CUDA_CHECK(cudaFree(d_pred_D));
+        return h_pred_I;
+    }
+
+    namespace Metrics {
+
+        /**
+        * @brief Compute the trustworthiness score
+        * @input param X: Data in original dimension
+        * @input param X_embedded: Data in target dimension (embedding)
+        * @input param n: Number of samples
+        * @input param m: Number of features in high/original dimension
+        * @input param d: Number of features in low/embedded dimension
+        * @input param n_neighbors: Number of neighbors considered by trustworthiness score
+        * @return Trustworthiness score
+        */
+        template<typename math_t>
+        double trustworthiness_score(const cumlHandle& h, math_t* X,
+            math_t* X_embedded, int n, int m, int d, int n_neighbors)
+        {
+            const int TMP_SIZE = MAX_BATCH_SIZE * n;
+
+            cudaStream_t stream = h.getStream();
+            auto alloc = h.getHostAllocator();
+
+            constexpr auto distance_type = MLCommon::Distance::DistanceType::EucUnexpandedL2Sqrt;
```

Comment:
This should be an argument to the function. We will be adding more distance types to knn soon.